util/linuxfw/nftables_runner.go

```diff
@@ -592,9 +592,23 @@ func New(logf logger.Logf, prefHint string) (NetfilterRunner, error) {
 	mode := detectFirewallMode(logf, prefHint)
 	switch mode {
 	case FirewallModeIPTables:
-		return newIPTablesRunner(logf)
+		// Note that we don't simply return an newIPTablesRunner here because it
+		// would return a `nil` iptablesRunner which is different from returning
+		// a nil NetfilterRunner.
+		ipr, err := newIPTablesRunner(logf)
```

Comment:
so isn't the caller (where's that?) also at fault, for ignoring the error result and blindly using the T of (T, error)? Unless a func documents that its T is meaningful, the assumption is it should be ignored on non-nil error.

Reply:
caller was doing this and then checking for `r.nfr == nil` later (shared/global object with lazy initialization), so not entirely wrong.
```
	var err error
	r.nfr, err = linuxfw.New(r.logf, r.netfilterKind)
        if err != nil { ... }
```